src/main/java/org/betonquest/betonquest/conversation/ConversationData.java

```diff
@@ -689,38 +704,47 @@ private class ConversationOption {
             this.pack = conversationID.getPackage();
             this.conversationName = conversationID.getBaseID();
             this.optionName = name;
             this.type = type;
             final ConfigurationSection conv = convSection.getConfigurationSection(type.getIdentifier() + "." + name);
 
             if (conv == null) {
-                pointers = new ArrayList<>();
-                extendLinks = new ArrayList<>();
                 return;
             }
 
             final String defaultLang = Config.getLanguage();
 
-            parsePrefix(name, type, conv, defaultLang);
-            parseText(name, type, conv, defaultLang);
-            parseConditions(name, type, conv);
-            parseEvents(name, type, conv);
-
-            pointers = Arrays.stream(GlobalVariableResolver.resolve(pack, conv.getString("pointers", conv.getString("pointer", ""))).split(","))
-                    .filter(StringUtils::isNotEmpty)
-                    .map(String::trim).toList();
-
-            extendLinks = Arrays.stream(GlobalVariableResolver.resolve(pack, conv.getString("extends", conv.getString("extend", ""))).split(","))
-                    .filter(StringUtils::isNotEmpty)
-                    .map(String::trim).toList();
+            try {
+                parsePrefix(name, type, conv, defaultLang);
+                parseText(name, type, conv, defaultLang);
+                parseConditions(name, type, conv);
+                parseEvents(name, type, conv);
+                parseStringToList(conv.getString("pointers", conv.getString("pointer", "")), pointers);
+                parseStringToList(conv.getString("extends", conv.getString("extend", "")), extendLinks);
+            } catch (final QuestRuntimeException e) {
+                throw new InstructionParseException(e);
+            }
+        }
+
+        private void parseStringToList(final String string, final List<String> list) throws InstructionParseException, QuestRuntimeException {
+            final String rawPointers = new VariableString(BetonQuest.getInstance().getVariableProcessor(), pack,
+                    string).getValue(null);
+            for (final String pointer : rawPointers.split(",")) {
+                if (!pointer.isEmpty()) {
+                    list.add(pointer.trim());
+                }
+            }
         }
 
-        private void parseEvents(final String name, final OptionType type, final ConfigurationSection conv) throws InstructionParseException {
+        private void parseEvents(final String name, final OptionType type, final ConfigurationSection conv) throws InstructionParseException, QuestRuntimeException {
             try {
-                for (final String rawEvent : GlobalVariableResolver.resolve(pack, conv.getString("events", conv.getString("event", ""))).split(",")) {
+                final String rawEvents = new VariableString(BetonQuest.getInstance().getVariableProcessor(), pack,
```

Comment:
You could put that repeated pattern in a new method and/or store that variable processor in a (local) field